packages/cli/cli/src/cli.ts

```diff
@@ -268,6 +269,53 @@ function addIrCommand(cli: Argv<GlobalCliOptions>, cliContext: CliContext) {
     );
 }
 
+function addRegisterCommand(cli: Argv<GlobalCliOptions>, cliContext: CliContext) {
+    cli.command(
+        ["register"],
+        "Register all generators in the specified group",
+        (yargs) =>
+            yargs
+                .option("version", {
+                    type: "string",
+                    description: "The version for the registered api",
+                })
+                .option("api", {
+                    string: true,
+                    description: "Only run the command on the provided API",
+                })
+                .option("token", {
+                    string: true,
+                    description: "The token for the organization",
+                }),
+        async (argv) => {
+            const project = await loadProjectAndRegisterWorkspaces(cliContext, {
+                commandLineWorkspace: argv.api,
+                defaultToAllWorkspaces: false,
+            });
+            const workspaces = project.workspaces;
+            if (workspaces === undefined || workspaces.length !== 1) {
+                cliContext.fail(`Failed to load ${argv.api}`);
+                return;
+            }
+            const workspace = workspaces[0];
+            if (workspace == null) {
+                cliContext.fail(`Failed to load ${argv.api}`);
+                return;
+            }
+            if (argv.token == null) {
+                cliContext.fail("Token must be specified");
+                return;
+            }
```

Comment:
@zachkirsch had to do a bunch of checks (want a single `Workspace` and a required `token`) is there a way around doing these checks manually?